WordPress/gutenberg
# HTML block: Fix accessibility issues on back-end

## What?


Fixes back-end accessibility of the HTML block.

## Why?


My PR does not fully fix this issue but I make good progress.

https://github.com/WordPress/gutenberg/issues/46846

## How?


- Manage focus when HTML toolbar button is selected.
- Manage focus when Preview button is selected.
- Announce via description instructions on how to use the preview when it is selected.

## Testing Instructions





1. Insert a HTML block.
2. Type some code.
3. Press Alt+F10 to focus the toolbar.
4. Select Preview.
5. Notice how focus moves to the block wrapper and you should be able to see the `aria-describedby` attribute which matches the ID of the block description.
6. Press Alt+F10.
7. Select HTML.
8. Focus should now be placed in the HTML text area.

### Notes

Wondering why the focus return from toolbar does not work for this block. Might be worth investigating this so focus does not need to be managed in the block itself.

### Testing Instructions for Keyboard


## Screenshots or screencast
